src/network/address.rs

```diff
@@ -95,12 +87,26 @@ impl Decodable for Address {
     fn consensus_decode<D: io::Read>(mut d: D) -> Result<Self, encode::Error> {
         Ok(Address {
             services: Decodable::consensus_decode(&mut d)?,
-            address: addr_to_be(Decodable::consensus_decode(&mut d)?),
+            address: read_be_address(&mut d)?,
             port: u16::swap_bytes(Decodable::consensus_decode(d)?)
         })
     }
 }
 
+/// Read a big-endian address from reader.
+fn read_be_address<R: io::Read>(mut r: R) -> Result<[u16; 8], encode::Error> {
+    let mut address = [0u16; 8];
+    let mut buf = [0u8; 2];
+
+    for i in 0..8 {
+        if io::Read::read(&mut r, &mut buf)? != 2 {
```

Comment:
This should be `read_exact`. `read` returning less bytes is not an error.

Reply:
Thanks man, I learned a bit more about readers.